sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/common/CounterSet.java

```diff
@@ -74,10 +74,10 @@ public void addNewCounter(Counter<?> counter) {
    * name but an incompatible kind had already been added
    */
   public synchronized <T> Counter<T> addOrReuseCounter(Counter<T> counter) {
-    Counter<?> oldCounter = counters.get(counter.getName());
+    Counter<?> oldCounter = counters.get(counter.getFlatName());
     if (oldCounter == null) {
       // A new counter.
-      counters.put(counter.getName(), counter);
+      counters.put(counter.getFlatName(), counter);
```

Comment:
Every one of these calls will need to get the synchronized lock currently. If we leave it synchronized, we should do `String flatName = counter.getFlatName()`.